src/Domain/Date/validateDateString.ts

```diff
@@ -0,0 +1,21 @@
+import dayjs = require('dayjs')
+import { ValidatedValue } from '../Validation/Validation'
+import { DateString } from './Date'
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/explicit-module-boundary-types
+export function validateDateString(date: any, format: string): ValidatedValue<DateString> {
```

Comment:
can we turn this into a class like `DateStringValidator` that implements `DateStringValidatorInterface` with a method `validate` ? 

All our projects are designed with the Dependency Injection pattern so it's easier to inject a given instance of a class and it's also easier to test/mock dependencies of them. If we're using bare functions it breaks the pattern and makes it hard to test.

Reply:
Also I think we can skip the date format validation from the request at this stage and add [Ajv](https://ajv.js.org/guide/formats.html#string-formats) at a later stage to handle that - wdyt?

Reply:
> can we turn this into a class like `DateStringValidator` that implements `DateStringValidatorInterface` with a method `validate` ?
> 
> All our projects are designed with the Dependency Injection pattern so it's easier to inject a given instance of a class and it's also easier to test/mock dependencies of them. If we're using bare functions it breaks the pattern and makes it hard to test.

We can turn it into a class, but I would rather not, as this would only add completely unnecessary complexity.

I get the general point and have no problem with DI, just there is nothing useful for it to do here.

Reply:
> Also I think we can skip the date format validation from the request at this stage and add Ajv at a later stage to handle that - wdyt?

I'll take it that our agreement [here](https://github.com/standardnotes/files/pull/3#discussion_r595219836) applies and we shall rework this validation later as we go along.